src/specialOffer/SpecialOffer.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+
+import { Button, ButtonStatus } from '../button'
+import { Disclaimer } from '../disclaimer'
+import { Paragraph } from '../paragraph'
+import { SubHeader } from '../subHeader'
+import {
+  StyledSpecialOffer,
+  StyledSpecialOfferButtonLine,
+  StyledSpecialOfferButtons,
+  StyledSpecialOfferContent,
+  StyledSpecialOfferFigure,
+  StyledSpecialOfferText,
+} from './SpecialOffer.style'
+
+type Picture = Readonly<{
+  url: string
+  alt: string
+}>
+
+type Link = Readonly<{
+  label: string
+  url: string
+}>
+
+type Button = Readonly<{
+  label: string
+  onClick: () => void
+}>
+
+type CTALinks = [] | [Link] | [Link, Link]
+type CTAButtons = [] | [Button] | [Button, Button]
+
+export type SpecialOfferProps = Readonly<{
+  picture: Picture
+  title: string
+  description: string
+  informationLink?: Link
+  actions?: CTALinks | CTAButtons
+}>
+
+export const SpecialOffer = ({
+  picture,
+  title,
+  description,
+  informationLink,
+  actions = [],
+}: SpecialOfferProps) => (
+  <StyledSpecialOffer>
+    <StyledSpecialOfferContent>
+      <StyledSpecialOfferFigure>
+        <img src={picture.url} alt={picture.alt} />
+      </StyledSpecialOfferFigure>
+    </StyledSpecialOfferContent>
+    <StyledSpecialOfferContent>
+      <StyledSpecialOfferText>
+        <SubHeader>{title}</SubHeader>
+        <Paragraph>{description}</Paragraph>
+        {informationLink && (
+          <Disclaimer useInfoIcon>
+            <Button status={ButtonStatus.UNSTYLED} href={informationLink.url}>
+              {informationLink.label}
+            </Button>
+          </Disclaimer>
+        )}
+      </StyledSpecialOfferText>
+      <StyledSpecialOfferButtons>
+        {(actions as Array<Link | Button>).map(action => (
```

Comment:
Doing so I'm forced to introduce a third (weaker) type because of 
![image](https://user-images.githubusercontent.com/748586/108509917-28469880-72be-11eb-907a-5ceac8ebf528.png)

I would need something like
```ts
type LinkOrButtonOrJustLabel = Readonly<{
  label: string
  url: string | undefined
  onClick: string | undefined
}>
```

Then map on the `actions` props to have an array of props for `<Button />`

```ts
  const buttonPropsList: Array<{
    label: string
    url: string | undefined
    onClick: string | undefined
  }> = (actions as  as Array<Link | Button>).map(action => ({ url: undefined, onClick: undefined, ...action })
```

The central issue here is me not wanting to expose a weak type `LinkOrButtonOrJustLabel` in the API of the component to be sure nobody pass just a label 🙄

I propose to stay like it is. At least we're hiding this design consideration inside the component, which is [good](https://prl.ccs.neu.edu/img/p-tr-1971.pdf)